lib/cucumber/formatter/io.rb

```diff
@@ -9,11 +11,15 @@ module Io
 
       def ensure_io(path_or_url_or_io)
         return nil if path_or_url_or_io.nil?
-        return path_or_url_or_io if path_or_url_or_io.respond_to?(:write)
-        io = if path_or_url_or_io.match(%r{^https?://})
-               HTTPIO.open(path_or_url_or_io)
+        return path_or_url_or_io if io?(path_or_url_or_io)
+
+        io = if url?(path_or_url_or_io)
+               url = path_or_url_or_io
+               reporter = url.start_with?(Cucumber::Cli::Options::CUCUMBER_PUBLISH_URL) ? URLReporter.new($stdout) : NoReporter.new
+               HTTPIO.open(url, nil, reporter)
              else
-               File.open(path_or_url_or_io, Cucumber.file_mode('w'))
+               path = path_or_url_or_io
```

Comment:
can remove this local assign to reduce complexity down

Reply:
done on master